cmd/eval-dev-quality/cmd/evaluate_test.go

```diff
@@ -613,4 +622,157 @@ func TestEvaluateExecute(t *testing.T) {
+				validateReportLinks(t, data, []string{"empty-response"})
+			},
+			filepath.Join("empty-response", "golang", "golang", "plain.log"): nil,
+		},
+	})
+
+	t.Run("Preloading", func(t *testing.T) {
+		{
+			// Setup provider and model mocking.
+			modelMock := modeltesting.NewMockModelNamed(t, "testing-provider/testing-model")
+			providerMock := providertesting.NewMockProviderNamedWithModels(t, "testing-provider", []model.Model{modelMock})
+			loaderMock := providertesting.NewMockLoader(t)
+			validate(t, &testCase{
+				Name: "Once for combined runs",
+
+				Before: func(t *testing.T, logger *log.Logger, resultPath string) {
+					modelMock.On("GenerateTestsForFile", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(metrics.Assessments{
+						metrics.AssessmentKeyProcessingTime: 1,
+					}, nil)
+					loaderMock.On("Load", "testing-provider/testing-model").Return(nil)
+					loaderMock.On("Unload", "testing-provider/testing-model").Return(nil)
+
+					provider.Register(
+						&struct {
+							provider.Provider
+							provider.Loader
+						}{
+							Provider: providerMock,
+							Loader:   loaderMock,
+						},
+					)
+				},
+				After: func(t *testing.T, logger *log.Logger, resultPath string) {
+					delete(provider.Providers, "testing-provider")
+
+					loaderMock.AssertNumberOfCalls(t, "Load", 1)
+					loaderMock.AssertNumberOfCalls(t, "Unload", 1)
+				},
+
+				Arguments: []string{
+					"--language", "golang",
+					"--model", "testing-provider/testing-model",
+					"--runs", "3",
+					"--runs-sequential",
+				},
+
+				ExpectedResultFiles: map[string]func(t *testing.T, filePath string, data string){
+					"categories.svg":    nil,
+					"evaluation.csv":    nil,
+					"evaluation.log":    nil,
+					"golang-summed.csv": nil,
+					"models-summed.csv": nil,
+					"README.md":         nil,
+					filepath.Join("testing-provider_testing-model", "golang", "golang", "plain.log"): nil,
+				},
+			})
+		}
+		{
+			// Setup provider and model mocking.
+			modelMock := modeltesting.NewMockModelNamed(t, "testing-provider/testing-model")
+			providerMock := providertesting.NewMockProviderNamedWithModels(t, "testing-provider", []model.Model{modelMock})
+			loaderMock := providertesting.NewMockLoader(t)
+			validate(t, &testCase{
+				Name: "Multiple times for interleaved runs",
+
+				Before: func(t *testing.T, logger *log.Logger, resultPath string) {
+					modelMock.On("GenerateTestsForFile", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(metrics.Assessments{
+						metrics.AssessmentKeyProcessingTime: 1,
+					}, nil)
+					loaderMock.On("Load", "testing-provider/testing-model").Return(nil)
+					loaderMock.On("Unload", "testing-provider/testing-model").Return(nil)
+
+					provider.Register(
+						&struct {
+							provider.Provider
+							provider.Loader
+						}{
+							Provider: providerMock,
+							Loader:   loaderMock,
+						},
+					)
```

Comment:
On #137 we moved the tests in `cmd/eval-dev-quality/cmd/evaluate_test.go` that were registering providers to `evaluate/evaluate_test.go` to avoid concurrency problems. I think we should do the same here.

Reply:
Thanks for mentioning it, that one totally slipped on me but is now fixed. Please check again.